### PR TITLE
Task7 post articles comments

### PR DESCRIPTION
**Pull request by kimberley-lee**
POST request responds with an object of specific properties:
- article_id
- comment_id
- votes
- created_at
- body
- author

handled errors
400- if user tries to post a comment without a specific property
400 - if user tries to post a comment and the value is in the incorrect format

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -207,4 +207,51 @@ describe("/api/articles/:article_id/comments", () => {
...
+          article_id: 2,
+          comment_id: 19,
+          votes: 0,
+          created_at: expect.any(String),
+          body: "I'm not actually a fan of cats.",
+          author: "butter_bridge",
+        });
+      });
+  });
+  test("POST 400: responds with an error message if user tries to post comment without body or author property", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:231]: excellent edge case

File: __tests__/app.test.js [hunk 2]
```diff
@@ -207,4 +207,51 @@ describe("/api/articles/:article_id/comments", () => {
...
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { message } = body;
+        expect(message).toBe("Bad request");
+      });
+  });
+  test("POST 400: responds with an error message if user tries to post in the incorrect format", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:244]: lovely

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -10,4 +10,14 @@ const getComments = (req, res, next) => {
     .catch(next);
 };
 
-module.exports = { getComments };
+const postComments = (req, res, next) => {
+  const { article_id } = req.params;
+  const { body, author } = req.body;
+  insertComments(article_id, body, author)
+    .then((comments) => {
+      res.status(201).send(comments);
```
Comment (roisinoreilly) [inline controllers/comments.controller.js:18]: comments needs to be sent back on an object

**Review by roisinoreilly — COMMENTED** [review]
great job so far! there's just a few extra tests I'd like you to add in, one for if you try to post with a username that isn't in the database, and a 404 non-existent id/400 invalid id still. Great edge cases though and excellent code!
